src/users/usersActions.js

```diff
@@ -35,6 +35,76 @@ export const reportPresence = (hasFocus: boolean = true, newUserInput: boolean =
   });
 };
 
+const typingWorkerBase = auth => ({
+  get_current_time: () => new Date().getTime(),
+
+  notify_server_start: user_ids_array => {
```

Comment:
These two functions look like they want explicit types on their arguments.

You may also want to have them return `Promise<void>` rather than `void` – not in this commit, of course, but possibly in the later refactoring step you mention. (I'm not sure whose responsibility error-reporting should be, but it seems like it should happen somewhere around here.)

Reply:
> These two functions look like they want explicit types on their arguments.

Sure.


> You may also want to have them return `Promise<void>` rather than `void` – not in this commit, of course, but possibly in the later refactoring step you mention. (I'm not sure whose responsibility error-reporting should be, but it seems like it should happen somewhere around here.)

Hmm, good thought. I'll have that in mind as I refactor on the webapp side.